x-pack/plugins/security_solution/public/common/components/ml_popover/hooks/use_enable_data_feed.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useState } from 'react';
+import { useAppToasts } from '../../../hooks/use_app_toasts';
+import { METRIC_TYPE, TELEMETRY_EVENT, track } from '../../../lib/telemetry';
+import { setupMlJob, startDatafeeds, stopDatafeeds } from '../api';
+import type { SecurityJob } from '../types';
+import * as i18n from './translations';
+
+// Enable/Disable Job & Datafeed -- passed to JobsTable for use as callback on JobSwitch
+export const useEnableDataFeed = () => {
+  const { addError } = useAppToasts();
+  const [isLoading, setIsLoading] = useState(false);
+
+  const enableDatafeed = async (job: SecurityJob, latestTimestampMs: number, enable: boolean) => {
```

Comment:
I think maybe this should be at least `useCallback`?
I'd suggest though just using React Query which solves the problem of calling external APIs and addresses a ton of edge cases. If you're curious ping us with @xcrzx and we can have a quick chat about what problems can be solved with it. @xcrzx talked about some of that at our On Week demos.

Reply:
Yes, I will use `useCallback` here.

I saw the On Week video and it is very interesting. I will make sure we consider React Query for future security solution refactors. Thanks for sharing it!